File: app/src/main/java/com/example/fellow_traveller/MessagesNotification/MyFirebaseIdService.java
package com.example.fellow_traveller.MessagesNotification;

import com.example.fellow_traveller.Models.GlobalClass;
import com.google.firebase.database.DatabaseReference;
import com.google.firebase.database.FirebaseDatabase;
import com.google.firebase.iid.FirebaseInstanceId;
import com.google.firebase.messaging.FirebaseMessagingService;

public class MyFirebaseIdService extends FirebaseMessagingService {
    private GlobalClass globalClass;
    private int myId;



    @Override
    public void onNewToken(String s) {
        super.onNewToken(s);
        //Retrieve current user's id
        globalClass = (GlobalClass) getApplicationContext();
        myId = globalClass.getCurrent_user().getId();

        String refreshToken = FirebaseInstanceId.getInstance().getToken();
        if(Integer.toString(myId)!=null){
            updateToken(refreshToken);
        }
    }

    private void updateToken(String refreshToken){
        //Retrieve current user's id
        globalClass = (GlobalClass) getApplicationContext();
        myId = globalClass.getCurrent_user().getId();

        DatabaseReference reference = FirebaseDatabase.getInstance().getReference("Tokens");
        Token token = new Token(refreshToken);
        reference.child(Integer.toString(myId)).setValue(token);

    }
}

File: app/src/main/java/gr/fellow/fellow_traveller/ui/home/chat/chat_notifications/MyFirebaseIdService.kt
package gr.fellow.fellow_traveller.ui.home.chat.chat_notifications

import com.google.firebase.database.FirebaseDatabase
import com.google.firebase.iid.FirebaseInstanceId
import com.google.firebase.messaging.FirebaseMessagingService
import dagger.hilt.android.AndroidEntryPoint
import javax.inject.Inject

@AndroidEntryPoint
class MyFirebaseIdService : FirebaseMessagingService() {
    @Inject
    lateinit var firebaseDatabase: FirebaseDatabase

    override fun onNewToken(p0: String) {
        super.onNewToken(p0)

        //Get users id
        var myId = "sdags"
        val refreshToken: String? = FirebaseInstanceId.getInstance().token
        //refreshToken = FirebaseMessaging.getInstance().token.toString()
        if (myId != null)
            refreshToken?.let { updateToken(it) }
    }

    fun updateToken(refreshToken: String) {
        //Get users id
        var myId = "sdags"

        val reference = firebaseDatabase.getReference("Tokens")
        val token = Token(refreshToken)
        reference.child(myId).setValue(token)

    }

}